gslib/tests/test_rpo.py

```diff
@@ -0,0 +1,187 @@
+      self.RunCommand('rpo', ['get', 'gs://invalid*'])
+
+  def test_set_called_with_incorrect_value_raises_error(self):
+    with self.assertRaisesRegexp(
+        CommandException,
+        r'Invalid value for rpo set. Should be one of \(ASYNC_TURBO\|DEFAULT\)'
+    ):
+      self.RunCommand('rpo', ['set', 'random', 'gs://boo*'])
+
+  def test_invalid_subcommand_raises_error(self):
+    with self.assertRaisesRegexp(
+        CommandException, 'Invalid subcommand "blah", use get|set instead'):
+      self.RunCommand('rpo', ['blah', 'DEFAULT', 'gs://boo*'])
+
+
+class TestRpo(testcase.GsUtilIntegrationTestCase):
+  """Integration tests for rpo command."""
+
+  # TODO: Delete this method once rpo get results are consistent
+  # from the backend, and replace this call with
+  # VerifyCommandGet(bucket_uri, 'rpo', 'DEFAULT').
+  # Currently, the rpo results are inconsistent
+  # and None is a valid default value for all the buckets.
+  # See b/197251750#comment19
+  # T
+  def _VerifyGetReturnsDefaultorNone(self, bucket_uri):
+    """Checks if the rpo get command returns default."""
+    try:
+      self.VerifyCommandGet(bucket_uri, 'rpo', 'DEFAULT')
+    except AssertionError:
+      self.VerifyCommandGet(bucket_uri, 'rpo', 'None')
+
+  @SkipForXML('RPO only runs on GCS JSON API')
+  def test_get_returns_default_for_dual_region_bucket(self):
+    bucket_uri = self.CreateBucket(location='us')
+    self._VerifyGetReturnsDefaultorNone(bucket_uri)
+
+  @SkipForXML('RPO only runs on GCS JSON API')
+  def test_get_returns_none_for_regional_bucket(self):
+    bucket_uri = self.CreateBucket(location='us-central1')
+    self.VerifyCommandGet(bucket_uri, 'rpo', 'None')
+
+  @SkipForXML('RPO only runs on GCS JSON API')
+  def test_set_and_get_async_turbo(self):
+    bucket_uri = self.CreateBucket(location='nam4')
+    self._VerifyGetReturnsDefaultorNone(bucket_uri)
+    self.RunGsUtil(['rpo', 'set', 'ASYNC_TURBO', suri(bucket_uri)])
+    self.VerifyCommandGet(bucket_uri, 'rpo', 'ASYNC_TURBO')
+
+  @SkipForXML('RPO only runs on GCS JSON API')
+  def test_set_default(self):
+    bucket_uri = self.CreateBucket(location='nam4')
+    self.RunGsUtil(['rpo', 'set', 'ASYNC_TURBO', suri(bucket_uri)])
+    self.VerifyCommandGet(bucket_uri, 'rpo', 'ASYNC_TURBO')
+    self.RunGsUtil(['rpo', 'set', 'DEFAULT', suri(bucket_uri)])
+    self._VerifyGetReturnsDefaultorNone(bucket_uri)
+
+  @SkipForXML('RPO only runs on GCS JSON API')
+  def test_set_async_turbo_fails_for_regional_buckets(self):
+    bucket_uri = self.CreateBucket(location='us-central1')
+    stderr = self.RunGsUtil(['rpo', 'set', 'ASYNC_TURBO',
+                             suri(bucket_uri)],
+                            expected_status=1,
+                            return_stderr=True)
+    self.assertIn('Invalid argument', stderr)
+
+  @SkipForJSON('Testing XML only behavior')
+  def test_xml_fails_for_set(self):
+    # use HMAC for force XML API
```

Comment:
Capitalize "use", and add period

Reply:
Done.

Reply:
Doesn't appear to be done

Reply:
Oops. I think the commit was not push. Really done now :)